addons/sale_timesheet/views/hr_timesheet_views.xml

```diff
@@ -7,21 +7,71 @@
             <field name="inherit_id" ref="hr_timesheet.hr_timesheet_line_search"/>
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='department_id']" position="after">
-                    <field name="timesheet_invoice_type"/>
+                    <field name="so_line"/>
                 </xpath>
                 <xpath expr="//filter[@name='month']" position="before">
-                    <filter name="billable_timesheet" string="Billable" domain="[('so_line', '!=', False)]"/>
-                    <filter name="non_billable_timesheet" string="Non Billable" domain="[('so_line', '=', False)]"/>
-                    <separator/>
                     <filter name="billable_time" string="Billed on Timesheets" domain="[('timesheet_invoice_type', '=', 'billable_time')]"/>
                     <filter name="billable_fixed" string="Billed at a Fixed Price" domain="[('timesheet_invoice_type', '=', 'billable_fixed')]"/>
                     <filter name="non_billable" string="Non Billable Tasks" domain="[('timesheet_invoice_type', '=', 'non_billable')]"/>
                     <separator/>
                 </xpath>
                 <xpath expr="//filter[@name='groupby_department']" position="before">
-                        <filter string="Billable Type" name="groupby_timesheet_invoice_type" domain="[]" context="{'group_by': 'timesheet_invoice_type'}"/>
+                    <filter string="Billable Type" name="groupby_timesheet_invoice_type" domain="[]" context="{'group_by': 'timesheet_invoice_type'}"/>
+                </xpath>
+                <xpath expr="//filter[@name='groupby_employee']" position="after">
+                    <filter string="Sales Order Item" name="groupby_sale_order_item" domain="[]" context="{'group_by': 'so_line'}"/>
+                    <filter string="Invoice" name="groupby_invoice" domain="[]" context="{'group_by': 'timesheet_invoice_id'}"/>
+                </xpath>
+            </field>
+    </record>
+
+    <record id="hr_timesheet_line_tree_inherit" model="ir.ui.view">
+        <field name="name">account.analytic.line.tree.inherit</field>
+        <field name="model">account.analytic.line</field>
+        <field name="inherit_id" ref="hr_timesheet.hr_timesheet_line_tree"/>
+        <field name="arch" type="xml">
+            <xpath expr="//tree/field[@name='unit_amount']" position="before">
+                <field name="so_line" optional="hide" options="{'no_create': True}"/>
```

Comment:
If the user has no access in Sales app, then I think he has not to go to the SOL form view.
See with PO. 🙂 

Moreover, we should filter the selection of possible SOL when the user want to change the SOL of a timesheet.
Like [here](https://github.com/odoo/odoo/blob/e865961df673bd2878f365b6daeb4a440dee7ac7/addons/sale_timesheet/views/project_task_views.xml#L151).
And also if the user edits the `so_line` field, we should set the `is_so_line_edited` to True. Normally with an onchange method, it will be good. The goal of `is_so_line_edited` field is to keep the SOL chosen by the user for this timesheet. So, if the user changed the SOL of a timesheet, then it is important to do this, to not change the `so_line` of this timesheet when the SOL in the task linked has changed.